src/sssom/util.py

```diff
@@ -299,6 +300,56 @@ def remove_mappings(self, msdf: "MappingSetDataFrame") -> None:
         self.df = self.df[self.df.columns.drop(list(self.df.filter(regex=r"_2")))]
         self.clean_prefix_map()
 
+    def propagate(self) -> List[str]:
+        """Propagate slot values from the set level down to individual records.
+
+        :return: The list of slots that were effectively propagated.
+        """
+        schema = SSSOMSchemaView()
+        propagated = []
+
+        for slot in schema.propagatable_slots:
+            if slot not in self.metadata:  # Nothing to propagate
+                continue
+            if slot in self.df.columns:  # Cannot propagate
+                continue
+
+            if schema.view.get_slot(slot).multivalued:
+                value = "|".join(self.metadata.pop(slot))
```

Comment:
when we're in dataframe world, are the lists really stored as strings that have been joined? this seems like something that should only happen during I/O

Reply:
> when we're in dataframe world, are the lists really stored as strings that have been joined?

Apparently so.

Try reading a SSSOM/TSV file that has multi-valued columns, you’ll find that in the dataframe returned by the `parse_sssom_table` function the values are stored as `|`-joined strings. Dealing with those values is apparently left to client code.